### ISSUE TITLE
Make construction of property dictionary lazy

### ISSUE THREAD

**Issue by ameligrana** [AntonOresten/DynamicStructs.jl#10]
Right now the dictionary is constructed when the instance is initialized which wastes some memory, I think we can do
better though, consider:

```julia
julia> using OrderedCollections

julia> # now
       struct A
           x::OrderedDict{Symbol, Any}
       end

julia> Base.summarysize(A(OrderedDict{Symbol, Any}()))
240


julia> # vs. 
        mutable struct P
           x::OrderedDict{Symbol, Any}
           P() = new()
       end

julia> struct B
           x::P
       end

julia> Base.summarysize(B(P()))
16

```

Then as soon as the first dynamic property is added, the dict is initialized.

Maybe this can cause some overhead, or maybe not.

**Comment by AntonOresten** [comment]
This isn't totally crazy. Here's a possible way to add it directly to the properties field with Refs:

```julia
julia> struct MyStruct
           _properties::Base.RefValue{Dict{Symbol, Any}}
           MyStruct() = new(Ref{Dict{Symbol, Any}}())
       end

julia> ms = MyStruct()
MyStruct(Base.RefValue{Dict{Symbol, Any}}(#undef))

julia> ms._properties[] = Dict{Symbol, Any}()
Dict{Symbol, Any}()
```

Your idea for a separate mutable properties struct looks waay cleaner though😅

**Comment by ameligrana** [comment]
I will give it a try with the mutable struct because I also like its cleanliness :-)

**Comment by AntonOresten** [comment]
Implemented in #11, merged in #12.
